src/main/java/com/dokany/java/DokanyMount.java

```diff
@@ -64,70 +62,79 @@ public long getVersion() {
+		this.mount(executor, () -> {});
+	}
+
+
+	/**
+	 * Calls {@link NativeMethods#DokanMain(DeviceOptions, DokanyOperations)} on the given executor and runs after mount termination a specified action.
+	 * No-op if this object is already mounted.
+	 * <p>
+	 * Additionally a shutdown hook invoking {@link #close()} is registered to the JVM.
 	 *
-	 * @Param Executor executor
+	 * @param executor
+	 * @param onDokanExit object with a run() method which is executed after the Dokan process exited
 	 */
-	public synchronized void mount(Executor executor) throws DokanyException {
+	public synchronized void mount(Executor executor, Runnable onDokanExit) throws DokanyException {
 		if (!isMounted) {
+			isMounted = true;
 			try {
 				Runtime.getRuntime().addShutdownHook(new Thread(this::close));
 
 				LOG.info("Dokany API/driver version: {} / {}", getVersion(), getDriverVersion());
 				//real mount op
-				mountFuture = CompletableFuture
-						.supplyAsync(() -> NativeMethods.DokanMain(deviceOptions, new DokanyOperationsProxy(fileSystem)), executor)
-						.handle((returnVal, exception) -> {
-							setIsMounted(false);
-							if (returnVal != null && returnVal.equals(0)) {
-								return 0;
+				CompletableFuture.supplyAsync(() -> NativeMethods.DokanMain(deviceOptions, new DokanyOperationsProxy(fileSystem)), executor)
+						.whenComplete((returnVal, throwable) -> {
+							isMounted = false;
+							onDokanExit.run();
+							if (throwable != null) {
+								throw new DokanyException(throwable);
 							}
-
-							if (returnVal == null) {
-								throw new DokanyException(exception);
-							} else {
-								throw new DokanyException("Return Code " + returnVal + ": " + MountError.fromInt(returnVal).getDescription());
+							if (returnVal != null && returnVal != 0) {
+								throw new DokanyException("Non-Zero return code " + returnVal + ": " + MountError.fromInt(returnVal).getDescription());
 							}
-						});
-
-				//check return value
-				try {
-					mountFuture.get(1000, TimeUnit.MILLISECONDS);
-				} catch (TimeoutException e) {
-					// up and running
-				} catch (ExecutionException e) {
-					LOG.error("Error while mounting", e);
-					if (e.getCause() instanceof DokanyException) {
-						throw (DokanyException) e.getCause();
-					} else {
-						throw new DokanyException(e);
-					}
-				} catch (InterruptedException e) {
-					Thread.currentThread().interrupt();
-				}
+						})
+						.get(1000, TimeUnit.MILLISECONDS);
 
-				setIsMounted(true);
+				//if the execution reaches this point, device was directly unmounted.
+				throw new DokanyException("Mount failed: Volume was instantly unmounted.");
 			} catch (UnsatisfiedLinkError err) {
-				LOG.error("Unable to load dokan driver.", err);
 				throw new DokanyException(err);
+			} catch (ExecutionException e) {
+				if (e.getCause() instanceof DokanyException) {
+					throw (DokanyException) e.getCause();
+				} else {
+					throw new DokanyException(e.getCause());
+				}
```

Comment:
Do we use Guava in this lib? If yes, we can simplify this to

```suggestion
				throwIfInstanceOf(e.getCause(), DokanyException.class);
				throw new DokanyException(e.getCause());
```